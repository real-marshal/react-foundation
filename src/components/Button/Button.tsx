import type { ReactNode } from 'react'
import styled from '@emotion/styled'

interface ButtonProps {
  children: ReactNode
}

const StyledButton = styled.button`
  border-radius: 10px;
  background-color: orange;
  padding: 0.5rem 1rem;
  border: none;
  cursor: pointer;
`

const Button = ({ children }: ButtonProps) => <StyledButton>{children}</StyledButton>

export { Button }
export type { ButtonProps }
