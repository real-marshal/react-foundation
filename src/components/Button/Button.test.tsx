import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import * as stories from './Button.stories'

const testStories = Object.values(composeStories(stories)).map(
  (Story) => [Story.storyName, Story] as [string, typeof Story]
)

describe('Button component', () => {
  it.each(testStories)('renders %s story', (_, Story) => {
    const view = render(<Story />)
    expect(view.baseElement).toMatchSnapshot()
  })
})
