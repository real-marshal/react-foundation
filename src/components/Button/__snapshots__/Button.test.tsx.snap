// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders Basic story 1`] = `
.emotion-0 {
  border-radius: 10px;
  background-color: orange;
  padding: 0.5rem 1rem;
  border: none;
  cursor: pointer;
}

<body>
  <div>
    <button
      class="emotion-0 emotion-1"
    >
      Button
    </button>
  </div>
</body>
`;
